src/main/java/com/pretz/dyntreeserver/UserService.java

```diff
@@ -14,16 +14,18 @@ public UserService(UserDao userDao) {
         this.userDao = userDao;
     }
 
-    public boolean checkIfUserExists(User user) {
-        if (userDao.exists(user.getId())) {
+    public boolean validateUser(User user) {
+        if (userDao.findByNameAndPassword(user.getName(), user.getPassword()).isPresent()) {
             return true;
         } else {
             return false;
         }
     }
 
     public boolean createUser(User user) {
-        userDao.save(user);
-        return checkIfUserExists(user);
+        if (!userDao.findByNameAndPassword(user.getName(), user.getPassword()).isPresent()) {
```

Comment:
isn't `name` unique?

Reply:
It is connected with previous remark, I guess. Yes, it should be unique. I will check if it is possible to create duplicate user in database via POST method in AuthController and, if not, think about adding some validation mechanism (or researching existing ones in Spring).